Tests/Test 5/wave_game.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include<queue>
+using namespace std;
+class Node{
+    public:
+    int x; // x coordainate
+    int y; // y coordainate
+    int dist; // distance from the start
+    Node(int x, int y, int distance){ //constructor
+        this->x=x;
+        this->y=y;
+        this->dist=distance;
+    }
+};
+int arr[500][500],n,m;
+int startX,startY,Distance; // the start coordinates and the distance
+bool used[500][500]; // boolean matrix where we will save if we used curtain vertex 
+long long int sum; // the sum of all vertices at the given distance from start
+queue<Node> q;
+void read(){
+    cin>>n>>m;
+    for(int i=0;i<n;i++){
+        for(int j=0;j<m;j++){
+            cin>>arr[i][j];
+        }
+    }
+    cin>>startX>>startY>>Distance;
+}
+bool isValid(int x, int y, int dist){
+     if(x>-1 && y>-1 && x<n && y<m && dist<=Distance && arr[x][y]!=-1 && used[x][y]==false) return true;
+     else return false;
+}
+void solve(){
+
+    q.push(Node(startX-1,startY-1,1)); // we start from (1,1) and it is at disatance 1 from itself
```

Comment:
It's better every node which is included at the queue to be marked as used, as otherwise there will be duplicates at the queue which is going to consume more time...